hardware_interface/test/test_robot_hardware_interface.cpp

```diff
@@ -0,0 +1,180 @@
+{
+  EXPECT_EQ(hw::HW_RET_OK, robot.register_joint_command_handle(&pos_command_handle));
+  EXPECT_EQ(hw::HW_RET_OK, robot.register_joint_state_handle(&state_handle));
+  EXPECT_EQ(hw::HW_RET_OK, robot.register_operation_mode_handle(&op_mode_handle));
+}
+
+TEST_F(TestRobotHardwareInterface, cannot_double_register_handles)
+{
+  SetUpHandles();
+
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.register_joint_command_handle(&pos_command_handle));
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.register_joint_state_handle(&state_handle));
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.register_operation_mode_handle(&op_mode_handle));
+}
+
+TEST_F(TestRobotHardwareInterface, can_get_registered_joint_names)
+{
+  SetUpHandles();
+  EXPECT_THAT(robot.get_registered_joint_names(), ElementsAre(JOINT_NAME));
+
+  SetUpNewHandles();
+  EXPECT_THAT(robot.get_registered_joint_names(), UnorderedElementsAre(JOINT_NAME, NEW_JOINT_NAME));
+}
+
+TEST_F(TestRobotHardwareInterface, can_get_registered_state_handles)
+{
+  SetUpHandles();
+  EXPECT_THAT(robot.get_registered_joint_state_handles(), SizeIs(1));
+
+  SetUpNewHandles();
+  EXPECT_THAT(robot.get_registered_joint_state_handles(), SizeIs(2));
+}
+
+TEST_F(TestRobotHardwareInterface, can_get_registered_command_handles)
+{
+  SetUpHandles();
+  EXPECT_THAT(robot.get_registered_joint_command_handles(), SizeIs(1));
+
+  SetUpNewHandles();
+  EXPECT_THAT(robot.get_registered_joint_command_handles(), SizeIs(2));
+}
+
+TEST_F(TestRobotHardwareInterface, can_get_registered_op_mode_handles)
+{
+  SetUpHandles();
+  EXPECT_THAT(robot.get_registered_operation_mode_handles(), SizeIs(1));
+
+  SetUpNewHandles();
+  EXPECT_THAT(robot.get_registered_operation_mode_handles(), SizeIs(2));
+}
+
+TEST_F(TestRobotHardwareInterface, can_get_handles_by_name)
+{
+  SetUpHandles();
+
+  const hw::JointStateHandle * state_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_joint_state_handle(JOINT_NAME, &state_handle));
+  state_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.get_joint_state_handle(NEW_JOINT_NAME, &state_handle));
+
+  hw::JointCommandHandle * cmd_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_joint_command_handle(JOINT_NAME, &cmd_handle));
+  cmd_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.get_joint_command_handle(NEW_JOINT_NAME, &cmd_handle));
+
+  hw::OperationModeHandle * op_mode_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_operation_mode_handle(JOINT_NAME, &op_mode_handle));
+  op_mode_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_ERROR, robot.get_operation_mode_handle(NEW_JOINT_NAME, &op_mode_handle));
+
+  SetUpNewHandles();
+  state_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_joint_state_handle(NEW_JOINT_NAME, &state_handle));
+  cmd_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_joint_command_handle(NEW_JOINT_NAME, &cmd_handle));
+  op_mode_handle = nullptr;
+  EXPECT_EQ(hw::HW_RET_OK, robot.get_operation_mode_handle(NEW_JOINT_NAME, &op_mode_handle));
+}
```

Comment:
Maybe also check handles are not null after successful `get_*_handle(...)`.

Reply:
good idea! added a new test for this